### ISSUE TITLE
Provide headers to send in an asset request

### ISSUE THREAD

**Issue by ninjasloth22** [sensu/sensu-go#3001]
It would be great if we could supply a hash or list of headers to send with the asset request.
This would give us more flexibility to where we can store assets and the types of authorization we could pass to the request. 

E.g
```
spec:
  headers:
    authorization: bearer lkajsd01823098sd
  url: https://someprotectedresouce.domain.com/asset1.tar.gz
```

**Comment by echlebek** [comment]
I like this :+1:

**Comment by ninjasloth22** [comment]
Was implemented in 5.11
https://docs.sensu.io/sensu-go/5.11/release-notes/#5-11-0-release-notes
